test/server/listener_manager_impl_test.cc

```diff
@@ -915,6 +916,45 @@ name: foo
   EXPECT_CALL(*listener_foo2, onDestroy());
 }
 
+TEST_F(ListenerManagerImplTest, BindToPortEqualToFalse) {
+  InSequence s;
+  EXPECT_CALL(*worker_, start(_));
+  manager_->startWorkers(guard_dog_);
+  const std::string listener_foo_yaml = R"EOF(
+name: foo
+address:
+  socket_address:
+    address: 127.0.0.1
+    port_value: 1234
+deprecated_v1:
+    bind_to_port: false
+filter_chains:
+- filters: []
+  )EOF";
+  Api::OsSysCallsImpl os_syscall;
+  auto syscall_result = os_syscall.socket(AF_INET, SOCK_STREAM, 0);
+  ASSERT_GE(syscall_result.rc_, 0);
+  ListenerHandle* listener_foo = expectListenerCreate(true, true);
+  EXPECT_CALL(listener_factory_, createListenSocket(_, _, _, false))
+      .WillOnce(
+          Invoke([this, &syscall_result](const Network::Address::InstanceConstSharedPtr& address,
+                                         Network::Address::SocketType socket_type,
+                                         const Network::Socket::OptionsSharedPtr& options,
+                                         bool bind_to_port) -> Network::SocketSharedPtr {
+            // When bind_to_port is equal to false, create socket fd directly, and do not get socket
```

Comment:
Can you add an explicit expectation with `Times(0)` to the mock hot restarter to make sure we never call it?